rustic-cargo.el

```diff
@@ -154,26 +154,34 @@ When calling this function from `rustic-popup-mode', always use the value of
 (defconst rustic-cargo-fn-regexp
   "^\s*\\(?:async\s+\\)?\s*fn\s+\\([^(]+\\)\s*(")
 
-(defun rustic-cargo--get-current-fn-fullname()
-  "Return full name of the fn around point including module name if any."
-  (let ((mod (rustic-cargo--get-current-mod))
-        (fn (rustic-cargo--get-current-fn-name)))
-    (if mod
-        (concat mod "::" fn)
-      fn)))
+(defun rustic-cargo--get-test-target()
+  "Return either a full fn name or a mod name, whatever is closer to the point."
+  (let ((mod-cons (rustic-cargo--get-current-mod))
+        (fn-cons (rustic-cargo--get-current-fn-name)))
+    (cond ((and mod-cons fn-cons)
+           ;; both conses contain (location . name)
+           (if (> (car mod-cons) (car fn-cons))
+               (cdr mod-cons)
+             (concat (cdr mod-cons) "::" (cdr fn-cons))))
+          (fn-cons (cdr fn-cons))
+          (t (cdr mod-cons)))))
 
 (defun rustic-cargo--get-current-mod ()
-  "Return mod name around point or nil."
+  "Return cons with location and mod name around point or nil."
   (save-excursion
-    (when (search-backward-regexp rustic-cargo-mod-regexp nil t)
-      (match-string 1))))
+    (progn
+      (goto-char (line-end-position))
+      (let ((location (search-backward-regexp rustic-cargo-mod-regexp nil t)))
```

Comment:
Maybe use `when-let` ?

Reply:
This is my first elisp code changes ever :) but the language looks pretty cool, I will have a look at improving that.

Reply:
Done. That looks quite good!